### PR TITLE
emulationstation-de: 3.0.2 -> 3.1.0

### PR DESCRIPTION
**Pull request by ftsimas**
[Changelog](https://gitlab.com/es-de/emulationstation-de/-/blob/master/CHANGELOG.md)

Commit [41fd33fd](https://gitlab.com/es-de/emulationstation-de/-/commit/41fd33fdc3dacef507b987ed316aec2b0d684317) has been backported to fix a buffer overflow detection by glibc's source fortification ([relevant issue here](https://gitlab.com/es-de/emulationstation-de/-/issues/1754)).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/em/emulationstation-de/package.nix [hunk 1]
```diff
@@ -37,36 +47,18 @@ stdenv.mkDerivation (finalAttrs: {
...
-    mkdir -p $out/share/es-de/
-    cp -r ../resources/ $out/share/es-de/resources/
-
-    # Desktop file
-    mkdir -p $out/share/applications
-    cp ../es-app/assets/org.es_de.frontend.desktop $out/share/applications/
-
-    # Icon
-    mkdir -p $out/share/icons/hicolor/scalable/apps
-    cp ../es-app/assets/org.es_de.frontend.svg $out/share/icons/hicolor/scalable/apps/
```
Comment (ivarmedi) [inline pkgs/by-name/em/emulationstation-de/package.nix:60]: I understand that the way it currently works is not ideal, but I think we still want to copy over the icon and desktop. I haven't really kept up with my Nixing lately, so please correct me if it's perhaps happening automatically?

Comment (ftsimas) [reply pkgs/by-name/em/emulationstation-de/package.nix:60]: In ES-DE's case yes, it does happen automatically.

CMake's setup hook sets the value of the `CMAKE_INSTALL_PREFIX` flag to the package's prefix:

https://github.com/NixOS/nixpkgs/blob/472d10b75bd462bb1e94625d23f9da5f15457d24/pkgs/by-name/cm/cmake/setup-hook.sh#L40-L42

<hr>

Which is the same flag that ES-DE uses to determine its install path:  
<details>
<summary>
<a href="https://gitlab.com/es-de/emulationstation-de/-/blob/v3.1.0/INSTALL.md?ref_type=tags&plain=1#L252-262">INSTALL.MD L252-262</a>
</summary>
<blockquote>
By default ES-DE will install under /usr on Linux and /usr/local on FreeBSD although this can be changed by setting the `CMAKE_INSTALL_PREFIX` variable.

The following example will build the application for installtion under /opt:

```
cmake -DCMAKE_INSTALL_PREFIX=/opt .
```

It's important to understand that this is not only the directory used by the install script, the CMAKE_INSTALL_PREFIX variable also modifies code inside ES-DE used to locate the required program resources. So it's necessary that the install prefix corresponds to the location where the application will actually be installed.

On Linux, if you're not building a package and instead intend to install using `make install` it's recommended to set the installation prefix to /usr/local instead of /usr.
</blockquote>
</details>

<hr>

And in that path it creates a `share` subfolder where the desktop file gets installed, along with its icon:
<details>
<summary>
<a href="https://gitlab.com/es-de/emulationstation-de/-/blob/v3.1.0/es-app/CMakeLists.txt?ref_type=tags#L291-321">es-app/CMakeLists.txt L291-321</a>
</summary>
<pre>
elseif(NOT ANDROID)
    install(TARGETS es-de RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
    install(TARGETS es-pdf-convert RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
    if(CMAKE_SYSTEM_NAME MATCHES Linux OR CMAKE_SYSTEM_NAME MATCHES FreeBSD)
        install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/assets/es-de.6.gz
            DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man6)
    else()
        install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/assets/es-de.6.gz
            DESTINATION ${CMAKE_INSTALL_PREFIX}/man/man6)
    endif()
    install(FILES ${CMAKE_SOURCE_DIR}/LICENSE
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/es-de)
    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/assets/org.es_de.frontend.desktop
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/assets/org.es_de.frontend.svg
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/pixmaps)
    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/assets/org.es_de.frontend.svg
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps)
    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/assets/org.es_de.frontend.appdata.xml
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/metainfo)
    install(DIRECTORY ${CMAKE_SOURCE_DIR}/licenses
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/es-de)
    install(DIRECTORY ${CMAKE_SOURCE_DIR}/themes/linear-es-de
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/es-de/themes)
    install(DIRECTORY ${CMAKE_SOURCE_DIR}/themes/modern-es-de
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/es-de/themes)
    install(DIRECTORY ${CMAKE_SOURCE_DIR}/themes/slate-es-de
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/es-de/themes)
    install(DIRECTORY ${CMAKE_SOURCE_DIR}/resources
        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/es-de)
endif()
</pre>
</details>

<hr>

In conclusion, the desktop file and the icon get copied as expected.

**Review by ivarmedi — APPROVED** [review]
Lgtm and thanks
